src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,90 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceTest {
+    private RegistrationServiceImpl registrationService;
+    private StorageDaoImpl storageDao;
+    private User user;
+
+    @BeforeEach
+    public void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+        user = new User("testuser", "password123", 20);
+        storageDao.add(user);
+    }
+
+    @Test
+    public void validate_AgeMoreThan18_Ok() {
+        User userOK = new User("11111111", "222222", 25);
+        assertDoesNotThrow(() -> registrationService.validate(userOK));
+    }
+
+    @Test
+    public void validate_AgeLessThan18_NotOk() {
+        user.setAge(0);
+        assertThrows(InvalidDataException.class, () -> registrationService.validate(user));
+        user.setAge(-100);
+        assertThrows(InvalidDataException.class, () -> registrationService.validate(user));
+        user.setAge(17);
+        assertThrows(InvalidDataException.class, () -> registrationService.validate(user));
+    }
+
+    @Test
+    public void validate_LoginMoreThan6_Ok() {
+        User userOK = new User("11111111", "222222", 25);
+        assertDoesNotThrow(() -> registrationService.validate(userOK));
+    }
+
+    @Test
+    public void validate_LoginLessThan6_NotOk() {
```

Comment:
```suggestion
    public void validate_LoginTooShort_NotOk() {
```
you may forget to change method name if constraint changes, better to be less specific